core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -772,6 +788,50 @@ public boolean load(String graphHopperFolder) {
         }
     }
 
+    private void checkProfilesConsistency() {
+        Set<String> profileSet = new LinkedHashSet<>(profiles.size());
+        for (ProfileConfig profile : profiles) {
+            boolean added = profileSet.add(profile.getName());
```

Comment:
you can add the profile directly to the set as the hashcode & equals is already implemented for that (?)

Or maybe shouldn't we do the check when calling setProfileConfigs or create a 
`LinkedHashMap<String, ProfileConfig> profiles` in the GraphHopper class on setProfileConfigs? This would also avoid the method getProfilesByName.

Reply:
Yes using a map is better, also because later when resolving the profile during a request we have to obtain the profile by name again. I used a map before and then changed it back to list, but now I do not remember why anymore -> its a map now :+1: